src/language-js/utils/index.js

```diff
@@ -1101,9 +1104,14 @@ const markerForIfWithoutBlockAndSameLineComment = Symbol(
   "ifWithoutBlockAndSameLineComment",
 );
 
-const isTSTypeExpression = createTypeCheckFunction([
+const isBinaryCastExpression = createTypeCheckFunction([
+  // TS
   "TSAsExpression",
   "TSSatisfiesExpression",
+  // Flow
+  "AsExpression",
+  "AsConstExpression",
```

Comment:
I'm not familiar with flow AST, any reason why we are not parsing it as `AsExpression` like TS? 
There should be edge cases format differently between Flow and TS.

Reply:
`x as const` is really just a one argument (`x`) operation. TS does something weird where they parse it as a binary `as` expression where the right hand side is a type identifier `const`, but then they match on it and ignore it?

Reply:
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions

Reply:
Make sense.

Reply:
`let x = "hello" as /* comment */ const;` format different. 😄 

Allow comment in "operator" also wired to me...

Reply:
Yeah I think it's weird that TS supports a comment in between `as` and `const` - it's just a side effect of how they decided to parse this